companion/src/simulation/telemetrysimu.cpp

```diff
@@ -1057,3 +1057,22 @@ void TelemetrySimulator::LogPlaybackController::setUiDataValues()
     }
   }
 }
+
+void TelemetrySimulator::startTelemetry()
+{
+  timer.start();
+}
+
+void TelemetrySimulator::stopTelemetry()
+{
+  timer.stop();
+
+  bool ok;
+  uint8_t buffer[FRSKY_SPORT_PACKET_SIZE] = {0};
+  generateSportPacket(buffer, ui->rssi_inst->text().toInt(&ok, 0) - 1, DATA_FRAME, RSSI_ID, 0);
+
+  if (ok) {
```

Comment:
Perhaps 
```
bool ok = (ui && ui->rssi_inst);
const int id = (ok ? ui->rssi_inst->text().toInt(&ok, 0) : 0);
if (!ok)
  return;
```
The pointer checks are a bit overly paranoid, but why not since it's called in destructor. But main point being the return before buffer creation/etc.

Reply:
My comment is actually more applicable to line 1070.